pkg/store/entities.go

```diff
@@ -2,54 +2,123 @@
 //
 // SPDX-License-Identifier: Apache-2.0
 
-// Package store contains implementation of the store for persisting pipeline entities
 package store
 
 import (
-	"context"
-	"github.com/onosproject/onos-api/go/onos/topo"
 	p4api "github.com/p4lang/p4runtime/go/p4/v1"
 )
 
-// EntityStore is an abstraction of a store capable of maintaining various P4 entities
-type EntityStore interface {
-	// ID returns the ID of the device whose control entities it persists.
-	ID() topo.ID
+/*
+Notes:
 
-	// Read accepts a query in form of a list of partially populated entities and returns any
-	// matching entities on the specified channel
-	Read(ctx context.Context, query []*p4api.Entity, ch chan<- p4api.Entity) error
+Create a map per device, per:
+- each table (map[tableId]*table; map[hash(entry.matches)]*entry)
+- all meters (map[meterId]*entry; same for all below)
```

Comment:
Indirect meter entries are identified by their index, so you probably need a nested map with index as a key pointing to the entry

Reply:
Yes. I've not yet decided whether to store each cell separately or as one map entry. Eitherway, yes, Index will be required to resolve the meter (and counters) to their individual cells.

Reply:
My plan is to implement table entries and counters first... the fan out and do the meters and the rest.